pkg/catalog/install_test.go

```diff
@@ -0,0 +1,167 @@
+package catalog_test
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+
+	helmv2 "github.com/fluxcd/helm-controller/api/v2beta1"
+	sourcev1 "github.com/fluxcd/source-controller/api/v1beta1"
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	profilesv1 "github.com/weaveworks/profiles/api/v1alpha1"
+
+	"github.com/weaveworks/pctl/pkg/catalog"
+	"github.com/weaveworks/pctl/pkg/catalog/fakes"
+)
+
+var _ = Describe("Install", func() {
+	var (
+		fakeHTTPClient *fakes.FakeHTTPClient
+	)
+
+	BeforeEach(func() {
+		fakeHTTPClient = new(fakes.FakeHTTPClient)
+		catalog.SetHTTPClient(fakeHTTPClient)
+	})
+
+	When("there is an existing catalog and user calls install for a profile", func() {
```

Comment:
can you add some tests for the more common error paths :smile:

Reply:
Yep! 👍